app/models/framework_response/array.rb

```diff
@@ -0,0 +1,29 @@
+class FrameworkResponse
+  class Array < FrameworkResponse
+    validates :value_text, absence: true
+    validates :value_json, presence: true, on: :update, if: -> { framework_question.required && parent.nil? }
```

Comment:
Maybe worth adding a method for this (in `FrameworkResponse` model) as it's repeated in a couple of these derived classes. Perhaps override this in descendant classes if needed?

```ruby
def requires_json_value?
  framework_question.required && parent.nil?
end
```

Reply:
I've moved all required validations to parent class